pandas/tests/frame/test_timeseries.py

```diff
@@ -323,6 +323,14 @@ def test_asfreq_datetimeindex(self):
         ts = df['A'].asfreq('B')
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        # GH 14340
+        expected = Series(index=pd.DatetimeIndex(
+            ["2016-09-29 11:00"])).asfreq('H')
+        result = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                        data=[3]).asfreq('H')
```

Comment:
@jorisvandenbossche Actually the issue was that we didn't set frequency properly for empty series. Hence I'm testing this particular thing by comparing with frequency of a non-empty `Series` that we make using similar parameters with an empty one.